protocols/peer/contracts/Peer.sol

```diff
@@ -50,10 +56,59 @@ contract Peer is IPeer, Ownable {
   ) public {
     swapContract = ISwap(_swapContract);
     if (_peerContractOwner != address(0)) {
-      transferOwnership(_peerContractOwner);
+      whitelist[_peerContractOwner] = true;
+      super.transferOwnership(_peerContractOwner);
     }
   }
 
+  /**
+    * @notice determines if an address to interact with this peer
+    * @param addressToCheck the address to check if whitelisted or not 
+    */
+  function isWhitelisted(address addressToCheck) external returns(bool) {
+    return whitelist[addressToCheck];
+  }
+
+  /**
+    * @notice adds to the list of whitelisted accounts that can interact with this peer
+    * @dev only callable by the owner of the contract
+    * @param addressToAdd the address to add to the whitelist
+    */
+  function addToWhitelist(address addressToAdd) external onlyOwner {
+    whitelist[addressToAdd] = true;
+    emit WhitelistAdded(addressToAdd);
+  }
+
+  /**
+    * @notice removes from the list of whitelisted accounts that can interact with this peer
+    * @dev only callable by the owner of the contract
+    * @param addressToRemove the address to add to the whitelist
+    */
+  function removeFromWhitelist(address addressToRemove) external onlyOwner {
+    require(addressToRemove != owner(), "OWNER_MUST_BE_WHITELISTED");
+    whitelist[addressToRemove] = false;
+    emit WhitelistRemoved(addressToRemove);
+  }
+
+  /**
+    * @notice transfers ownership to the new owner and ensures that whitelist is updated
+    * @dev only callable by the owner of the contract
+    * @param newOwner the address of the new owner of the contract
+    */
+  function transferOwnership(address newOwner) public onlyOwner {
+    whitelist[newOwner] = true;
+    whitelist[owner()] = false;
+    super.transferOwnership(newOwner);
+  }
+
+  /**
+    * @dev only whitelisted ensures that only whitelisted parties can call the method it modifies
+    */
+  modifier onlyWhitelisted() {
```

Comment:
put modifier at the top of the function
also pretty confident you can just call require(whitelist[msg.sender] , "CALLER..")

Reply:
you mean top of the contract @poiresel ?
agree with the removal of `== true`